pkg/container/README.md

```diff
@@ -0,0 +1,216 @@
+# Containers for TiFlow
+
+## ChunkQueue
+
+An memory efficient queue implementation in Golang.
+
+### Features
+
+- Support for Generics (Go 1.18+)
+- Chunked memory allocation. Compared to slice, it's GC-friendly and free of 
+redundant copy when pushing and popping
+- Support iterator operations and multiple range methods
+
+### Design
+The design of ChunkQueue is as follows.
+ChunkQueue divides the memory allocation into equal-length segments. A segment
+is called a chunk. ChunkQueue maintains an array []*chunks, addresses of chunks
+in use.
+
+The length of each chunk, `chunkLength`, is determined during the queue's initialization.
+For example, the chunk length of a ChunkQueue[T] is `max{16, 1Kb / size of type T}`.
+For each chunk, interval [l, r) are valid elements.
+
+Push() puts an element at the free index of the last chunk. A new chunk will be
+added if the last chunk is complete. Pop() pops an element from the first non-empty
+index of the first chunk. The whole chunk will be poped if it becomes empty.
+
+PushMany() and PopMany(int) operates multiple elements at a time, optimized by
+copy(). The second return value of PopMany(n int) indicates whether the number
+of elements popped meets the input n.
+
+For a ChunkQueue, there is at most one non-full chunk at both ends.
+
+
+![ChunkQueue Design](./queue/ChunkQueue.jpg)
```

Comment:
Can we put it into `https://github.com/pingcap/tiflow/tree/master/docs/media`?

Reply:
Or you can use the URL for it. For example: copy it into a comment you'll get a GitHub image URl.

Reply:
now in /docs/media